### PR TITLE
Adds eslint configuration and refactor project and files

### PR DESCRIPTION
**Pull request by josemigallas**
TODO:

- [x] Include all files into eslint check
- [x] Fix build

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1,35 +1,31 @@
...
-    return res.sendStatus(404);
-});
+App.get("*", (_, res) => res.sendStatus(404));
 
 exports.run = function (callback) {
     server = App.listen(port, () => callback(App));
 };
 
-exports.stop = function() {
+exports.stop = function () {
```
Comment (cfoskin) [inline server.js]: @josemigallas  should this be () =>  also?

Comment (josemigallas) [reply server.js]: Yes thanks, I definitely left some files unreviewed.

File: .babelrc [hunk 2]
```diff
@@ -1,4 +1,4 @@
 {
-  "presets": ["env", "react"],
+  "presets": ["env", "react", "es2015"],
```
Comment (pb82) [inline .babelrc]: what does `es2015` give us that is not already in `react`?

File: package.json [hunk 3]
```diff
@@ -1,40 +1,52 @@
...
+    "license": "Apache-2.0",
+    "repository": "https://github.com/aerogear/data-sync-ui",
+    "scripts": {
+        "lint": "eslint server test ui webpack.config.js --fix",
+        "server": "webpack --mode development && nodemon --watch ui --exec babel-node server/index.js",
+        "build": "webpack --mode production",
+        "clean": "rm -rf dist public/js/bundle.js node_modules",
+        "start": "node ./index.js",
+        "test": "NODE_ENV=test mocha",
+        "docker:build": "docker build -t aerogear/data-sync-ui:latest .",
```
Comment (pb82) [inline package.json]: needs a rebase, this file hase changed

File: .babelrc (hunk 2 above)
Comment (josemigallas) [reply .babelrc]: It used to fix some vscode error I think, but I'll remove it.

File: package.json (hunk 3 above)
Comment (josemigallas) [reply package.json]: should be fixed now

File: package.json [hunk 4]
```diff
@@ -3,11 +3,14 @@
...
   "description": "AeroGear Sync Server UI",
   "main": "index.js",
-  "license": "MIT",
+  "license": "Apache-2.0",
+  "repository": "https://github.com/aerogear/data-sync-ui",
   "scripts": {
-    "server": "webpack --mode development && nodemon --watch ui index.js",
+    "lint": "eslint server test ui webpack.config.js --fix",
+    "server": "webpack --mode development && nodemon --watch ui --exec babel-node server/index.js",
     "build": "webpack --mode production",
```
Comment (pb82) [inline package.json]: don't we have to take `babel-node` into account here? In the docker container we are not using babel, we need to provide something that can be started with pure node.

Comment (josemigallas) [reply package.json]: Yes that's another lost during rebase, it's fixed now.

**Pull request merged**